docs/tutorials/using_datasets/IQL_torchrl.py

```diff
@@ -0,0 +1,522 @@
+
+# %%
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+base_env = GymEnv(env_id, device=device)
+env = TransformedEnv(base_env, DoubleToFloat())
+
+# %%
+# Building a replay buffer
+# ~~~~~~~~~~~~~~~~~~~~~~~~
+
+# %%
+# The Minari dataset we will be using is `pen-human-v1 <https://minari.farama.org/main/datasets/pen/human/>`_, which consists of 25 human demonstrations. We can create a replay buffer using ``MinariExperienceReplay()``:
+
+# %%
+from torchrl.data.datasets.minari_data import MinariExperienceReplay
+from torchrl.data.replay_buffers import SamplerWithoutReplacement
+
+dataset_id = "pen-human-v1"
+batch_size = 256
+
+replay_buffer = MinariExperienceReplay(
+    dataset_id,
+    split_trajs=False,
+    batch_size=batch_size,
+    sampler=SamplerWithoutReplacement(),
+    transform=DoubleToFloat(),
+)
+
+# %%
+# **Note:** We add the transform ``DoubleToFloat()`` so that the step data is consistent with our environment.
+#
+# On the first run, the dataset will be downloaded from Farama servers and stored in the local cache directory (e.g. ``~/.cache/torchrl/minari/``). Once the dataset is loaded we can iterate over the replay buffer or use ``replay_buffer.sample()`` to load batches of transitions.
+
+# %%
+# Implicit Q-Learning
+# ~~~~~~~~~~~~~~~~~~~
+
+# %%
+# For completeness, we give a quick overview of `Implicit Q-Learning (IQL) <https://arxiv.org/abs/2110.06169>`_ and how it tries to tackle some of the challenges of offline RL. Those who are familiar with IQL or are only interested in the practical implementation can skip to the next section: :ref:`label-defining_the_model`.
+#
+# The main challenge in offline RL is **distribution shift**: Function approximators (e.g. for the Q-function) are trained on one distribution of data, the offline dataset, but are evaluated on another distribution, that of the newly trained policy. When evaluating state-action pairs well outside of the original distribution, they may extrapolate poorly, resulting in a policy that performs well on the dataset but poorly in practice. To make this more precise, consider an offline dataset :math:`\mathcal{D} = \{(s_t, a_t, r_t, s_{t+1}), \dots\}`. A standard starting point for offline RL algorithms is minimising the temporal difference error, by optimizing the objective
+#
+# .. math::
+#   L_{\rm TD}(\theta) = \mathbb{E}_{(s, a, s') \sim \mathcal{D}} \left[ \left( r(s, a) + \gamma \max_{a'} Q_{\hat{\theta}}(s', a') - Q_\theta(s, a) \right)^2 \right]
+#
+#
+# .. figure:: /_static/img/tutorials/IQL_torchrl_expectile_example.png
+#    :width: 250
+#    :figwidth: 300
+#    :align: right
+#
+#    Expectiles of an example conditional distribution :math:`y \sim m_\tau(s)`. For :math:`\tau = 0.5` the expectile is the mean while for :math:`\tau \approx 1` it approximates the maximum of the distribution `(Kostrikov et al 2021) <https://arxiv.org/abs/2110.06169>`_.
```

Comment:
Will this be displayed in this position? It looks like it breaks the flow

Reply:
When rendered it floats to the right so it doesn't disrupt the text. However, if you have a better position in mind I would be happy to move it.